### PR TITLE
WIP: Initial add of OLM manifests

### PR DESCRIPTION
**Pull request by djzager**
A WIP PR to create a CSV for the cluster-network-addons-operator.

TODO:
- [ ] Provide a means to test this PR
- [ ] Nail down the rules needed for successful operation of the cluster-network-addons-operator (can't count on being able to bind to a named cluster-role)

Fixes: https://github.com/kubevirt/cluster-network-addons-operator/issues/38

### CODE REVIEW

**Comment by djzager** [comment]
@phoracek If you could have a look and tell me if I am on the right track for this effort.

**Comment by djzager** [comment]
I may need to keep #32 in mind when working on this.

**Comment by phoracek** [comment]
Thanks @djzager, will take a look. Btw, what is the icon? :D

File: deploy/olm-catalog/kubevirt-cluster-network-addons/0.0.1/kubevirt-cluster-network-addons-operator.v0.0.1.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+            "namespace":"kubevirt-network-addons"
+          },
+          "spec": {
+            "multus":{},
+            "linuxBridge":{},
+            "sriov":{},
+            "kubeMacPool": {
+              "startPoolRange": "02:00:00:00:00:00",
+              "endPoolRange": "FD:FF:FF:FF:FF:FF",
+              "imagePullPolicy": "Always",
```
Comment (phoracek) [inline deploy/olm-catalog/kubevirt-cluster-network-addons/0.0.1/kubevirt-cluster-network-addons-operator.v0.0.1.clusterserviceversion.yaml]: is not part of kubeMacPool section, but the whole spec

File: deploy/olm-catalog/kubevirt-cluster-network-addons/0.0.1/kubevirt-cluster-network-addons-operator.v0.0.1.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    capabilities: "Full Lifecycle"
+    categories: "Virtualization"
+    alm-examples: |
+      [
+        {
+          "apiVersion":"networkaddonsoperator.network.kubevirt.io/v1alpha1",
+          "kind":"NetworkAddonsConfig",
+          "metadata": {
+            "name":"kubevirt-network-addons",
+            "namespace":"kubevirt-network-addons"
```
Comment (phoracek) [inline deploy/olm-catalog/kubevirt-cluster-network-addons/0.0.1/kubevirt-cluster-network-addons-operator.v0.0.1.clusterserviceversion.yaml]: NetworkAddonsConfig is not a namespaced resource, please drop the namespace

File: deploy/olm-catalog/kubevirt-cluster-network-addons/0.0.1/kubevirt-cluster-network-addons-operator.v0.0.1.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+  annotations:
+    capabilities: "Full Lifecycle"
+    categories: "Virtualization"
+    alm-examples: |
+      [
+        {
+          "apiVersion":"networkaddonsoperator.network.kubevirt.io/v1alpha1",
+          "kind":"NetworkAddonsConfig",
+          "metadata": {
+            "name":"kubevirt-network-addons",
```
Comment (phoracek) [inline deploy/olm-catalog/kubevirt-cluster-network-addons/0.0.1/kubevirt-cluster-network-addons-operator.v0.0.1.clusterserviceversion.yaml]: we recognize only one name, "cluster", everything else will be ignored

File: deploy/olm-catalog/kubevirt-cluster-network-addons/0.0.1/kubevirt-cluster-network-addons-operator.v0.0.1.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,288 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: kubevirt-cluster-network-addons-operator.v0.0.1
+  namespace: placeholder
+  annotations:
+    capabilities: "Full Lifecycle"
```
Comment (phoracek) [inline deploy/olm-catalog/kubevirt-cluster-network-addons/0.0.1/kubevirt-cluster-network-addons-operator.v0.0.1.clusterserviceversion.yaml]: what does this mean?

File: deploy/olm-catalog/kubevirt-cluster-network-addons/0.0.1/kubevirt-cluster-network-addons-operator.v0.0.1.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+      ]
+    description: Additional networking components for Kubernetes
+spec:
+  displayName: KubeVirt Cluster Network Addons
+  description: |
+    Deploy additional networking components for Kubernetes
+  keywords:
+    - KubeVirt
+    - Virtualization
+    - Networking
```
Comment (phoracek) [inline deploy/olm-catalog/kubevirt-cluster-network-addons/0.0.1/kubevirt-cluster-network-addons-operator.v0.0.1.clusterserviceversion.yaml]: Please add
- Multus
- CNI
- macpool
- SR-IOV
- Bridge

**Comment by phoracek** [comment]
Can we wait for the manifest generator to be merged and use it? I think it would really help to make versions parametrized. BTW we keep released manifests in https://github.com/kubevirt/cluster-network-addons-operator/releases. Is there any reason to keep "0.0.1" in source code?

**Comment by phoracek** [comment]
Nevermind the PR, let's do this one and I replace it later with manifests generated in code.

**Comment by djzager** [comment]
Closing this in favor of implementing a manifest generator.

**Pull request closed without merging**